src/ruby/yast/profiler.rb

```diff
@@ -0,0 +1,58 @@
+
+require "yast"
+
+module Yast
+  class Profiler
+    class << self
+      include Yast::Logger
+
+      RESULT_PATH = "/tmp/yast_profile.txt"
+
+      # Start the Ruby Profiler. It start profilling. It also disables ruby VM
+      # optimizations, so code execution will be slower.
+      def start
+        raise "multiple profiller start detected" if @started
+        @original_compile_options = RubyVM::InstructionSequence.compile_option
+        @started = true
+        require "profiler"
+
+        # turn on tracing and turn of specialized instruction to get complete profilling
```

Comment:
It is specially optimized common instructions, it is described in book "Ruby Under a Microscope: Learning Ruby Internals Through Experiment" I am not sure if it is described somewhere in publicly available web page. So it basically turn off VM optimization which will hide some ruby methods.